server/client.go

```diff
@@ -3603,7 +3606,11 @@ func (c *client) setHeader(key, value string, msg []byte) []byte {
 	// Put the original message back.
 	// FIXME(dlc) - This is inefficient.
 	bb.Write(msg[omi:])
-	nsize := bb.Len() - LEN_CR_LF
+	nsize := bb.Len()
```

Comment:
Would reverse and set to -LEN and add back in when MQTT.

Reply:
I can do that. Right now, I am not using setHeader() with an MQTT client, so I could even completely remove the change, but if one day I use that, I would get parse errors on the other side if I don't have such code.

Reply:
I am saying default single op should be normal NATS case and MQTT case can put it back vs 3 ops for normal NATS flow.

Reply:
Oh sure.. I was just saying that I could even remove the changes in this function since I am not using it. But I will make the changes you mention and keep the MQTT case just in case I use it in the future.